### PR TITLE
📖 Update links that were pointing to the `master` branch of `amphtml` and associated repos

### PR DESCRIPTION
**Pull request by rsimha**
This PR updates all links in `amphtml` documentation that were pointing to the `master` branch of the `amphtml` (and associated) repos to use `main` instead. It can be merged at leisure after the default branch of `amphtml` is renamed (see [this plan](https://docs.google.com/document/d/1BN8CkM3b2nydENRI2X1tE6nCxcaeVliUIaC2e-mXZbg/edit#)), because GitHub links are auto-redirected.

Sending this out for review and owners approval. Won't merge until the branch rename is done. Default branches of associated repos like `amp.dev`, `worker-dom`, `meta`, `wg-*`, etc. have already been renamed.

Partial fix for #32195

### CODE REVIEW

File: build-system/tasks/create-golden-css/css/main.css [hunk 1]
```diff
@@ -146,49 +146,42 @@ html.i-amphtml-ios-embed.i-amphtml-ios-overscroll > #i-amphtml-wrapper {
...
 
-
 .i-amphtml-element {
   display: inline-block;
 }
 
 .i-amphtml-layout-nodisplay,
-[layout=nodisplay]:not(.i-amphtml-layout-nodisplay)
-{
+[layout='nodisplay']:not(.i-amphtml-layout-nodisplay) {
```
Comment (kristoferbaxter) [inline build-system/tasks/create-golden-css/css/main.css:154]: I'm a bit surprised to see these changes in this PR, but they are non-breaking a look like prettier reformatting.

File: build-system/tasks/create-golden-css/css/main.css [hunk 2]
```diff
@@ -307,16 +299,14 @@ i-amphtml-sizer {
...
   position: relative;
   overflow: hidden !important;
   color: transparent !important;
 }
 
-.i-amphtml-notbuilt:not(.i-amphtml-layout-container) > * ,
-[layout]:not([layout=container]):not(.i-amphtml-element) > *
-{
+.i-amphtml-notbuilt:not(.i-amphtml-layout-container) > *,
+[layout]:not([layout='container']):not(.i-amphtml-element) > * {
```
Comment (kristoferbaxter) [inline build-system/tasks/create-golden-css/css/main.css:309]: This change should be functionally equivalent, but do we want it?

**Review by kristoferbaxter — APPROVED** [review]
Thanks for getting this cleaned up!

File: build-system/tasks/create-golden-css/css/main.css (hunk 1 above)
Comment (rsimha) [reply build-system/tasks/create-golden-css/css/main.css]: Reverted this change.

**Review by danielrozenberg — DISMISSED** [review]
Blocking comment per @rsimha's request

**Review by lannka — APPROVED** [review]

**Comment by rsimha** [comment]
@ampproject/wg-caching Could I get an owners approval for this PR? (The branch has been renamed, all this PR does is update existing links to point to the new branch.)

**Review by dvoytenko — APPROVED** [review]

**Review by Gregable — APPROVED** [review]

**Pull request merged**